serializer.rst

```diff
@@ -2,223 +2,443 @@ How to Use the Serializer
 =========================
 
 Symfony provides a serializer to serialize/deserialize to and from objects and
```

Comment:
I'd start the article with a focus on why/when to use this utility.

This is the current intro:

```
Symfony provides a serializer to serialize/deserialize to and from objects and
different formats (e.g. JSON or XML).
```

Summarized to the max, it's saying that _"Symfony includes a serializer so you can serialize/deserialize"_. But for newcomers, what's exactly to serialize/deserialize?

Maybe we can mention some of its common use-cases:

```
Symfony provides a serializer to transform data structures from one format to another.
For example, to transform a collection of PHP arrays into a JSON structure before
sending it as a response in your API; to transform an incoming JSON payload back
to that collection of PHP arrays; to map configuration data from XML to YAML or
vice versa; etc.
```